src/core/jbig2.js

```diff
@@ -1293,8 +1394,718 @@ var Jbig2Image = (function Jbig2ImageClosure() {
     onImmediateLosslessHalftoneRegion() {
       this.onImmediateHalftoneRegion.apply(this, arguments);
     },
+    onTables(currentSegment, data, start, end) {
+      let customTables = this.customTables;
+      if (!customTables) {
+        this.customTables = customTables = {};
+      }
+      customTables[currentSegment] = decodeTablesSegment(data, start, end);
+    },
   };
 
+  function HuffmanLine(lineData) {
+    if (lineData.length === 2) {
+      // OOB line.
+      this.isOOB = true;
+      this.prefixLength = lineData[0];
+      this.prefixCode = lineData[1];
```

Comment:
initialize all fields (rangeLow, rangeLength, isLowerRange) here and in the same order as below. add `this.isOOB = false;` below too

Reply:
Done.